services/users-service/src/main/java/za/co/protogen/config/SecurityConfiguration.java

```diff
@@ -0,0 +1,28 @@
+package za.co.protogen.config;
+
+import jakarta.ws.rs.HttpMethod;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableWebSecurity
+@EnableMethodSecurity(prePostEnabled = true)
+public class SecurityConfiguration {
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http
+                .authorizeHttpRequests(authorize -> authorize
+                        .requestMatchers("/users/**",HttpMethod.GET).authenticated()
```

Comment:
Here you need to permit the POST request and authenticate every other request. That includes the PUT and DELETE requests.